tests/README.md

```diff
@@ -1,24 +1,100 @@
 # Integration tests
 
-Integration tests require a running Kubevirt cluster.  Once you have a running
-Kubevirt cluster, you can use the `-master` and the `-kubeconfig` flags to
-point the tests to the cluster.
+Kubevirt integration tests are a suite of end-to-end tests which run
+against an active Kubevirt cluster.
 
-## Run them on an arbitrary KubeVirt installation
+The tests may execute against various clusters, from integrated kubevirtci
+deployments and up to external kubernetes or OpenShift ones.
+
+## Kubevirtci
+
+Kubevirtci provides containerized Kubernetes clusters which can be used
+to test KubeVirt and similar projects.
 
+KubeVirt integration tests are being run against kubevirtci clusters on CI
+and is the simplest way to run the tests locally in a development environment.
+
+To run the test locally:
+- Configure the desired cluster setup (provider, number of nodes and nics).
+  Note: Some tests may run on on specific configuration and HW.
 ```
-cd tests # from the git repo root folder
-go test -kubeconfig=path/to/my/config -config=default-config.json
+export KUBEVIRT_PROVIDER=k8s-1.17
+export KUBEVIRT_NUM_NODES=2
+export KUBEVIRT_NUM_SECONDARY_NICS=2
 ```
 
-## Run them on one of the core KubeVirt providers
+- Raise a cluster based on the configuration.
+  At this stage, it will not contain kubevirt.
+  ```make cluster-up```
+- Build KubeVirt from sources.
+  ```make```
+- Deploy KubeVirt onto the cluster.
+  ```make cluster-sync```
+- Run the tests.
+  ```make functest```
+- Take down and delete the cluster.
+  ```make cluster-down```
+
+On a regular workflow of development or debugging, rebuilding KubeVirt is
+needed only when the production code is changed.
+In case only test code changes, `make functest` will rebuild the test code
+implicitly.
+
+## Existing (external) Kubevirt Cluster
+
+The tests may be executed against an already deployed (and active) Kubevirt
+cluster.
+
+The environment needs to be adjusted to allow the tests to operate agains
+such a cluster.
+The following high level steps are required to accomplish this:
+- Clone kubevirt repository on the machine that has access to the cluster.
+- Identify the deployed kubevirt version.
+- From the git repo, checkout to the commit that corrolates to the running
+kubevirt cluster.
+- Run the tests by executing the `run-tests-on-external-cluster.sh`.
```

Comment:
please add path, and possible a link, as this is the first time you mention this script and it's not yet clear what it is

Reply:
Done